include/irr/asset/format/encodePixels.h

```diff
@@ -499,87 +499,87 @@ namespace irr { namespace video
     {
         int8_t* pix = reinterpret_cast<int8_t*>(_pix);
         for (uint32_t i = 0u; i < 3u; ++i)
-            pix[i] = _input[i] * 127.;
+            pix[i] = static_cast<int8_t>(_input[i] * 127.);
     }
 	
     template<>
     inline void encodePixels<asset::EF_R8G8B8_USCALED, double>(void* _pix, const double* _input)
     {
         uint8_t* pix = reinterpret_cast<uint8_t*>(_pix);
         for (uint32_t i = 0u; i < 3u; ++i)
-            pix[i] = _input[i];
+            pix[i] = static_cast<int8_t>(_input[i]);
```

Comment:
this bad-boy needs a `core::clamp`

Reply:
Actuallly I don't get it technically. I mean, won't there be a maximum value of `int8_t` if I casted `double` to `int8_t` assuming my double variable is around 100k? Won't it be clamped automaticaly even though?

Reply:
Nah, it's UB. See [c++14 standard](http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2014/n4296.pdf), paragraph 4.9.1.
However on gcc indeed there is a clamp http://coliru.stacked-crooked.com/a/9f6f5ec4b0ded22c
But ofc you cannot rely on behavior of single compiler since standard says that anything really can happen (just as well it could be wrap-around did in some tricky way or some random shit)

Reply:
> Nah, it's UB. See [c++14 standard](http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2014/n4296.pdf), paragraph 4.9.1.
> However on gcc indeed there is a clamp http://coliru.stacked-crooked.com/a/9f6f5ec4b0ded22c
> But ofc you cannot rely on behavior of single compiler since standard says that anything really can happen (just as well it could be wrap-around did in some tricky way or some random shit)

Thank you so much, I'm gonna check it!